File: app/src/main/java/com/autohub/skln/activities/user/StudentClassSelect.java
package com.autohub.skln.activities.user;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.util.SparseArray;
import android.view.View;
import android.widget.ImageView;

import androidx.annotation.NonNull;

import com.autohub.skln.BaseActivity;
import com.autohub.skln.R;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.firestore.SetOptions;

import java.util.HashMap;
import java.util.Map;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper;

import static com.autohub.skln.utills.AppConstants.CLASS_1;
import static com.autohub.skln.utills.AppConstants.CLASS_10;
import static com.autohub.skln.utills.AppConstants.CLASS_11;
import static com.autohub.skln.utills.AppConstants.CLASS_12;
import static com.autohub.skln.utills.AppConstants.CLASS_2;
import static com.autohub.skln.utills.AppConstants.CLASS_3;
import static com.autohub.skln.utills.AppConstants.CLASS_4;
import static com.autohub.skln.utills.AppConstants.CLASS_5;
import static com.autohub.skln.utills.AppConstants.CLASS_6;
import static com.autohub.skln.utills.AppConstants.CLASS_7;
import static com.autohub.skln.utills.AppConstants.CLASS_8;
import static com.autohub.skln.utills.AppConstants.CLASS_9;
import static com.autohub.skln.utills.AppConstants.KEY_STDT_CLASS;
import static com.autohub.skln.utills.AppConstants.KEY_USER_ID;

public class StudentClassSelect extends BaseActivity {

    @BindView(R.id.ivClass1)
    ImageView ivClass1;

    @BindView(R.id.ivClass2)
    ImageView ivClass2;

    @BindView(R.id.ivClass3)
    ImageView ivClass3;

    @BindView(R.id.ivClass4)
    ImageView ivClass4;

    @BindView(R.id.ivClass5)
    ImageView ivClass5;

    @BindView(R.id.ivClass6)
    ImageView ivClass6;

    @BindView(R.id.ivClass7)
    ImageView ivClass7;

    @BindView(R.id.ivClass8)
    ImageView ivClass8;

    @BindView(R.id.ivClass9)
    ImageView ivClass9;

    @BindView(R.id.ivClass10)
    ImageView ivClass10;

    @BindView(R.id.ivClass11)
    ImageView ivClass11;

    @BindView(R.id.ivClass12)
    ImageView ivClass12;

    private String selectedClass;
    private SparseArray<ImageView> ivClasses = new SparseArray<>();
    private SparseArray<String> CLASSES = new SparseArray<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_student_class_select);
        ButterKnife.bind(this);

        ivClass1.setEnabled(false);
        ivClass2.setEnabled(false);
        ivClass3.setEnabled(false);
        ivClass4.setEnabled(false);
        ivClass5.setEnabled(false);
        ivClass6.setEnabled(false);
        ivClass7.setEnabled(false);
        ivClass8.setEnabled(false);
        ivClass9.setEnabled(false);
        ivClass10.setEnabled(false);
        ivClass11.setEnabled(false);
        ivClass12.setEnabled(false);

        ivClasses.put(R.id.LLClass1, ivClass1);
        ivClasses.put(R.id.LLClass2, ivClass2);
        ivClasses.put(R.id.LLClass3, ivClass3);
        ivClasses.put(R.id.LLClass4, ivClass4);
        ivClasses.put(R.id.LLClass5, ivClass5);
        ivClasses.put(R.id.LLClass6, ivClass6);
        ivClasses.put(R.id.LLClass7, ivClass7);
        ivClasses.put(R.id.LLClass8, ivClass8);
        ivClasses.put(R.id.LLClass9, ivClass9);
        ivClasses.put(R.id.LLClass10, ivClass10);
        ivClasses.put(R.id.LLClass11, ivClass11);
        ivClasses.put(R.id.LLClass12, ivClass12);

        CLASSES.put(R.id.LLClass1, CLASS_1);
        CLASSES.put(R.id.LLClass2, CLASS_2);
        CLASSES.put(R.id.LLClass3, CLASS_3);
        CLASSES.put(R.id.LLClass4, CLASS_4);
        CLASSES.put(R.id.LLClass5, CLASS_5);
        CLASSES.put(R.id.LLClass6, CLASS_6);
        CLASSES.put(R.id.LLClass7, CLASS_7);
        CLASSES.put(R.id.LLClass8, CLASS_8);
        CLASSES.put(R.id.LLClass9, CLASS_9);
        CLASSES.put(R.id.LLClass10, CLASS_10);
        CLASSES.put(R.id.LLClass11, CLASS_11);
        CLASSES.put(R.id.LLClass12, CLASS_12);
    }

    @OnClick(R.id.btnNext)
    public void onNextClick() {

        if (TextUtils.isEmpty(selectedClass)) {
            showSnackError(R.string.select_class_text_1);
            return;
        }

        showLoading();

        final boolean isSeniorClass = selectedClass.equalsIgnoreCase(CLASS_11) || selectedClass.equalsIgnoreCase(CLASS_12);

        Map<String, Object> user = new HashMap<>();
        String uid = getFirebaseAuth().getCurrentUser().getUid();
        user.put(KEY_STDT_CLASS, selectedClass);
        user.put(KEY_USER_ID, uid);

        getFirebaseStore().collection(getString(R.string.db_root_students)).document(uid).set(user, SetOptions.merge())
                .addOnSuccessListener(new OnSuccessListener<Void>() {
                    @Override
                    public void onSuccess(Void aVoid) {
                        hideLoading();
                        Intent i = new Intent(StudentClassSelect.this, StudentHey.class);
                        i.putExtra("is_senior", isSeniorClass);
                        startActivity(i);
                    }
                })
                .addOnFailureListener(new OnFailureListener() {
                    @Override
                    public void onFailure(@NonNull Exception e) {
                        hideLoading();
                        showSnackError(e.getMessage());
                    }
                });
    }

    @OnClick({R.id.LLClass1, R.id.LLClass2, R.id.LLClass3, R.id.LLClass4, R.id.LLClass5, R.id.LLClass6, R.id.LLClass7, R.id.LLClass8, R.id.LLClass9, R.id.LLClass10, R.id.LLClass11, R.id.LLClass12})
    public void onClassItemClick(View view) {
        for (int i = 0; i < ivClasses.size(); i++) {
            int key = ivClasses.keyAt(i);
            ImageView ivClass = ivClasses.get(key);

            if (key == view.getId()) {
                ivClass.setEnabled(true);
                selectedClass = CLASSES.get(key);
            } else {
                ivClass.setEnabled(false);
            }
        }
    }

    @Override
    protected void attachBaseContext(Context newBase) {
        super.attachBaseContext(CalligraphyContextWrapper.wrap(newBase));
    }
}

File: app/src/main/java/com/autohub/skln/activities/user/StudentClassSelect.kt
package com.autohub.skln.activities.user

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import com.autohub.skln.BaseActivity
import com.autohub.skln.R
import com.autohub.skln.activities.user.fragments.StudentClassFragment
import com.autohub.skln.databinding.ActivityStudentClassSelectBinding
import com.autohub.skln.listeners.ClassSelectionListner
import com.autohub.skln.models.Classdata
import com.autohub.skln.utills.AppConstants.*
import com.google.firebase.firestore.SetOptions
import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper
import java.util.*
import kotlin.collections.ArrayList
import kotlin.collections.indices
import kotlin.collections.set


class StudentClassSelect : BaseActivity(), ClassSelectionListner {
    override fun selectedClass(position: Int, isSecondSelected: Boolean, selectedClass: String) {
        this.selectedClass = selectedClass
        for (i in fragmentsList.indices) {
            if (position != i) {
                fragmentsList[i].updateFragment(true)
            } else {
                fragmentsList[i].updateFragment(isSecondSelect = isSecondSelected)
            }
        }
    }


    private var mBinding: ActivityStudentClassSelectBinding? = null
    private var classdatalist: ArrayList<Classdata> = ArrayList()
    private var fragmentsList: ArrayList<StudentClassFragment> = ArrayList()
    private var selectedClass: String = ""

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_student_class_select)
        mBinding!!.callback = this
        insertClassData()

        var countList: ArrayList<String> = ArrayList()
        countList.add("1")
        countList.add("2")
        countList.add("3")
        countList.add("4")
        countList.add("5")
        countList.add("6")
        getFragments(countList)
        var pagerAdapter = PagerAdapter(supportFragmentManager, fragmentsList)
        mBinding!!.viewpager.adapter = pagerAdapter
        mBinding!!.viewpager.offscreenPageLimit = 5
        mBinding!!.wormDotsIndicator.setViewPager(mBinding!!.viewpager)


    }

    private fun getFragments(countList: ArrayList<String>) {

        for (position in countList.indices) {

            if (position == 0) {
                fragmentsList.add(StudentClassFragment.newInstance(position, ArrayList(classdatalist.subList(0, 2))))

            } else if (position == 1)
                fragmentsList.add(StudentClassFragment.newInstance(position, ArrayList(classdatalist.subList(2, 4))))
            else if (position == 2)
                fragmentsList.add(StudentClassFragment.newInstance(position, ArrayList(classdatalist.subList(4, 6))))
            else if (position == 3)
                fragmentsList.add(StudentClassFragment.newInstance(position, ArrayList(classdatalist.subList(6, 8))))
            else if (position == 4)
                fragmentsList.add(StudentClassFragment.newInstance(position, ArrayList(classdatalist.subList(8, 10))))
            else
                fragmentsList.add(StudentClassFragment.newInstance(position, ArrayList(classdatalist.subList(10, 12))))
        }
    }

    private fun insertClassData() {
        classdatalist.add(Classdata(R.color.black, R.drawable.one, false, CLASS_1))
        classdatalist.add(Classdata(R.color.two, R.drawable.two, false, CLASS_2))
        classdatalist.add(Classdata(R.color.three, R.drawable.three, false, CLASS_3))
        classdatalist.add(Classdata(R.color.four, R.drawable.four, false, CLASS_4))
        classdatalist.add(Classdata(R.color.five, R.drawable.one, false, CLASS_5))
        classdatalist.add(Classdata(R.color.six, R.drawable.one, false, CLASS_6))
        classdatalist.add(Classdata(R.color.seven, R.drawable.one, false, CLASS_7))
        classdatalist.add(Classdata(R.color.eight, R.drawable.one, false, CLASS_8))
        classdatalist.add(Classdata(R.color.nine, R.drawable.one, false, CLASS_9))
        classdatalist.add(Classdata(R.color.ten, R.drawable.one, false, CLASS_10))
        classdatalist.add(Classdata(R.color.eleven, R.drawable.one, false, CLASS_11))
        classdatalist.add(Classdata(R.color.tweleve, R.drawable.one, false, CLASS_12))
    }


    override fun attachBaseContext(newBase: Context) {
        super.attachBaseContext(CalligraphyContextWrapper.wrap(newBase))
    }


    inner class PagerAdapter(fragmentManager: FragmentManager, fragmentsList: ArrayList<StudentClassFragment>) :
            FragmentStatePagerAdapter(fragmentManager, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

        // 2
        override fun getItem(position: Int): Fragment {
            return fragmentsList[position]


        }

        override fun getPageWidth(position: Int): Float {
            return 0.5f
        }

        // 3
        override fun getCount(): Int {
            return fragmentsList.size
        }
    }

    fun onNextClick() {

        if (TextUtils.isEmpty(selectedClass)) {
            showSnackError(R.string.select_class_text_1)
            return
        }

        showLoading()

        val isSeniorClass = selectedClass.equals(CLASS_11, ignoreCase = true) || selectedClass.equals(CLASS_12, ignoreCase = true)

        val user = HashMap<String, Any>()
        val uid = firebaseAuth.currentUser!!.uid
        user[KEY_STDT_CLASS] = selectedClass
        user[KEY_USER_ID] = uid

        firebaseStore.collection(getString(R.string.db_root_students)).document(uid).set(user, SetOptions.merge())
                .addOnSuccessListener {
                    hideLoading()
                    val i = Intent(this@StudentClassSelect, StudentHey::class.java)
                    i.putExtra("is_senior", isSeniorClass)
                    startActivity(i)
                }
                .addOnFailureListener { e ->
                    hideLoading()
                    showSnackError(e.message)
                }
    }
}
